rotkehlchen/chain/aggregator.py

```diff
@@ -1486,6 +1496,10 @@ def detect_evm_accounts(
 
         return added_accounts
 
+    def iterate_supported_chains(self) -> Iterator[SUPPORTED_CHAIN_IDS]:
+        """Iterate the supported chains"""
+        yield from get_args(SUPPORTED_CHAIN_IDS)
```

Comment:
Hm this is not what I meant exactly, though close.  Just iterating the supported chains makes no sense for it to be part of the chains aggregator.

What does make sense it to do this iteration but yield the supported chain evm manager. Which you will then user outside.

notice to drop the supported from the function name as it;s implicit due to being part of chain aggregator.

```suggestion
    def iterate_evm_chain_managers(self) -> Iterator['EvmManager']:
        """Iterate the supported evm chain managers"""
        for chain_id in get_args(SUPPORTED_CHAIN_IDS):
            yield self.get_evm_manager(chain_id)
```

and then use this in the loop you did in rotkehlchen.py

Reply:
oh ok yes makes sense I will adjust it :)